### ISSUE TITLE
Duplicate same shapes/strokes prevents to fill (vector level)

### ISSUE THREAD

**Issue by openanim** [opentoonz/opentoonz#2159]
### Issue Summary

I often duplicate strokes / shapes in the same place, however OT will prevent filling in this case. As if in a certain way the duplicate was not exactly at the same value Z. and it must be put back in the same place to fill it, a little like the groups.

 But the logical behavior would be that it fills the form since it is supposed to be the same. This is also probably part of a wider problem of strokes overlap with the fill. 

### Steps to Reproduce
On vector level
Draw a cercle, copy paste by using selection tool
use shit arrow to reposition the drawing in the same place and try to fill.

### Expected Results
The shape have to be filled
Even if they are two circles, it is expected that the one above could fill even if the one below would remain hidden.

**Comment by jpturcotte** [comment]
Could you post a screengrab, please? [ScreenToGif](https://www.screentogif.com/) works well for that, and it is free. Thank you!

**Comment by beeheemooth** [comment]
@openanim  strange, but i can't reproduce this problem.

![ezgif-2-0cc44bff0b](https://user-images.githubusercontent.com/25724133/42727952-31351b36-87b8-11e8-9374-cd311021366a.gif)

**Comment by beeheemooth** [comment]
oops, it works randomly 

![ezgif-2-7b0edb008b](https://user-images.githubusercontent.com/25724133/42728059-6c37120a-87ba-11e8-9df3-7a892abdc9f8.gif)

**Comment by RodneyBaker** [comment]
If you immediately Group the shape after copy/paste does this problem still persist?

**Comment by melieconiek** [comment]
I do a lot of layer merging and will sometimes merge a copy of linework over top of existing linework accidentally. It's one of the fill bugs not fixed by a level reload.

**Comment by artisteacher** [comment]
If the pasted drawing is grouped right after pasting and before setting it back in the original position, filling does work. 

Interestingly, if you fill the shape before copying & pasting and then return it to perfectly overlap the original, the fill is removed from both shapes. Moving the pasted drawing away allows it to be filled again. The original drawing stays unfillable. At this point, using "convert to simplified vectors" on the cell, ends up with 2 slightly different drawings instead of identical. Setting those slightly different shapes to overlap allows filling to work. In this case, filling while the 2 shapes are overlapping, overrides the original fill color on the bottom drawing if there is one. 

In case it's related, I also ended up with a couple crashes while experimenting with this, with the latest nightly build & MacOS 10.13.5.
[crash1.txt](https://github.com/opentoonz/opentoonz/files/2650194/crash1.txt)
[crash2.txt](https://github.com/opentoonz/opentoonz/files/2650195/crash2.txt)

**Comment by openanim** [comment]
yes I know this workaround but I prefer to shift a bit to allow filling because the groups are another source of filling problem.

**Comment by RodneyBaker** [comment]
Revisiting this older report with a few thoughts on issues with color fill...

> groups are another source of filling problem

Groups are a key element of vector fill; a requirement even.
As such they will likely be a core part of the final solution.
From the Opentoonz documentation:

> As groups define which strokes belong to a certain plane, when using the Fill tool (fill), only areas defined by strokes **_within the same group_** can be filled.

(emphasis added)

One thing I've been hoping to explore is how ungrouping everything might help in alleviating vector fill problems.
The same can be said for grouping everything but my understanding is that if everything is ungrouped all objects belong to the same plane/group.  This suggests we can maximize color fill success by using either extreme... group everything we wish to color fill or ungroup everything.**

Currently we can see one aspect of this 'zero groups' approach that leads to more reliable color fills when we use the option to 'Replace vectors with Simplified Vectors'.  If I understand correctly, a side effect of this process is that assigned groups are discarded and therefore all strokes belong to the same 'group'.

The primary downside of this 'Simplify Vectors' process is of course that it segmentizes all the vector lines/strokes at intersections which will lead to other potential problems if the user wishes to use autoinbetweening etc..

**I am not aware of a current process that allows users to 'Ungroup All' short of as a side effect of other process such as via 'Simplify Vectors'.

**Comment by andeon** [comment]
I’d like to know what the conclusion of this issue is, since we already have autogroup and autofill.

**Comment by RodneyBaker** [comment]
Converting to discussion as the originator is a ghost.
